tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11458,48 +11497,39 @@ void hlsl::DiagnoseTranslationUnit(clang::Sema *self) {
     }
   }
 
-  // Validate that there is no recursion; start with the entry function.
-  // NOTE: the information gathered here could be used to bypass code generation
-  // on functions that are unreachable (as an early form of dead code
-  // elimination).
-  if (pEntryPointDecl) {
-    const auto *shaderModel =
-        hlsl::ShaderModel::GetByName(self->getLangOpts().HLSLProfile.c_str());
+  FunctionDecl *pResult = ValidateNoRecursion(self, pEntryPointDecl);
```

Comment:
nit: I have a strong aversion to introducing new code that uses Hungarian notation, and so does the LLVM style guide. This code is pretty mixed in how it names things, I don’t expect that you’ll fix that, but lets at least add new things with more style-consistent names.

https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly

Reply:
"nit:" and "strong aversion" give some pretty divergent messages here. I am emphatically indifferent. If this were to diverge from its immediately approximate Hungarian brethren, it would stand out, but would be loved just the same. If it chose to go with the flow of its surroundings, I would understand completely.

Reply:
You say it would stand out, but we're actually very inconsistent in this function already. We have `pResult` and `pEntryPointDecl`, but `shaderModel`, `Attr`, `NL`, and `self`. The changed code alone mixes at least 3 different schemes for naming variables.

We should at least make sure that the changed code is consistent with itself.

Reply:
I decided to use camelCase. Would be cool if clang-format could auto-format variable names, but I recognize that's a hard problem 😐

Reply:
`clang-tidy` does support enforcing variable naming conventions, but it doesn't support applying the changes only to changed regions, so you really have to get large portions of the codebase to comply to use it effectively.